projects/igniteui-angular/src/lib/chips/chip.spec.ts

```diff
@@ -97,377 +104,310 @@ describe('IgxChip', () => {
+        });
 
-        expect(firstComponent.componentInstance.displayDensity).toEqual(DisplayDensity.comfortable);
+        it('should make chip cosy when density is set to cosy', () => {
+            const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+            const fourthComponent = components[3];
 
-        // Assert default css class is applied
-        const comfortableComponents = fix.debugElement.queryAll(By.css('.igx-chip'));
+            expect(fourthComponent.componentInstance.displayDensity).toEqual(DisplayDensity.cosy);
 
-        expect(comfortableComponents.length).toEqual(2);
-        expect(comfortableComponents[0].nativeElement).toBe(firstComponent.nativeElement);
-    });
+            // Assert cosy css class is added
+            const cosyComponents = fix.debugElement.queryAll(By.css(CHIP_COSY_CLASS));
 
-    it('should make chip comfortable when density is set to comfortable', () => {
-        const fix = TestBed.createComponent(TestChipComponent);
-        fix.detectChanges();
+            expect(cosyComponents.length).toEqual(1);
+            expect(cosyComponents[0].nativeElement).toBe(fourthComponent.nativeElement);
+        });
 
-        const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
-        const secondComponent = components[1];
+        it('should set correctly color of chip when color is set through code', () => {
+            const chipColor = 'rgb(255, 0, 0)';
 
-        expect(secondComponent.componentInstance.displayDensity).toEqual(DisplayDensity.comfortable);
+            const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+            const firstComponent = components[0];
+            const chipAreaElem = firstComponent.queryAll(By.css(CHIP_ITEM_CLASS))[0];
 
-        // Assert default css class is applied
-        const comfortableComponents = fix.debugElement.queryAll(By.css('.igx-chip'));
+            firstComponent.componentInstance.color = chipColor;
 
-        expect(comfortableComponents.length).toEqual(2);
-        expect(comfortableComponents[1].nativeElement).toBe(secondComponent.nativeElement);
+            expect(chipAreaElem.nativeElement.style.backgroundColor).toEqual(chipColor);
+            expect(firstComponent.componentInstance.color).toEqual(chipColor);
+        });
     });
 
-    it('should make chip compact when density is set to compact', () => {
-        const fix = TestBed.createComponent(TestChipComponent);
-        fix.detectChanges();
+    describe('Interactions Tests: ', () => {
+        beforeEach(() => {
+            fix = TestBed.createComponent(TestChipComponent);
+            fix.detectChanges();
+        });
 
-        const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
-        const thirdComponent = components[2];
+        it('should not trigger onRemove event when a chip is focused and delete button is pressed when not removable', () => {
```

Comment:
It would be better to be remove that the chip is being focused from the title since these few test no longer include element focusing .

Reply:
@skrustev I updated the tests descriptions.